### PR TITLE
[DOC] Update INSTALL.md

### PR DESCRIPTION
**Pull request by chenhong805**
Add note for build option `-DOPENTELEMETRY_INSTALL`.

Fixes # (issue):

https://github.com/open-telemetry/opentelemetry-cpp/issues/2573

## Changes

Added documentation for build arg `-DOPENTELEMETRY_INSTALL`.

### CODE REVIEW

File: INSTALL.md [hunk 1]
```diff
@@ -85,6 +85,7 @@ You can link OpenTelemetry C++ SDK with libraries provided in
    - `-DWITH_OTLP_GRPC=ON` : To enable building OTLP GRPC exporter.
    - `-DWITH_OTLP_HTTP=ON` : To enable building OTLP HTTP exporter.
    - `-DWITH_PROMETHEUS=ON` : To enable building prometheus exporter.
+   - [`-DOPENTELEMETRY_INSTALL=ON`](https://github.com/open-telemetry/opentelemetry-cpp/blob/main/sdk/CMakeLists.txt): To install `libopentelemetry_common.so` needed for external code linking that rely on `ld`.
```
Comment (lalitb) [inline INSTALL.md:88]: This CMake option is not just to install libopentelemetry_common.so, but to install all the built otel-cpp libraries

Comment (ThomsonTan) [reply INSTALL.md:88]: I think link to the source code is not needed here.

Comment (chenhong805) [reply INSTALL.md]: Updated.
Removed code link, and updated description.

**Comment by esigo** [comment]
@chenhong805 any update on CLA?

**Comment by esigo** [comment]
closing as there was no reponse on CLA.

File: INSTALL.md [hunk 2]
```diff
@@ -85,6 +85,7 @@ You can link OpenTelemetry C++ SDK with libraries provided in
    - `-DWITH_OTLP_GRPC=ON` : To enable building OTLP GRPC exporter.
    - `-DWITH_OTLP_HTTP=ON` : To enable building OTLP HTTP exporter.
    - `-DWITH_PROMETHEUS=ON` : To enable building prometheus exporter.
+   - `-DOPENTELEMETRY_INSTALL=ON`: To install `otel-cpp` library needed for external code linking that rely on `ld`.
```
Comment (lalitb) [inline INSTALL.md:88]:
```suggestion
   - `-DOPENTELEMETRY_INSTALL=ON`: To install `otel-cpp` library needed for external code linking.
```

nit - remove `ld` to keep instructions cross-platform.

File: INSTALL.md [hunk 3]
```diff
@@ -85,6 +85,7 @@ You can link OpenTelemetry C++ SDK with libraries provided in
    - `-DWITH_OTLP_GRPC=ON` : To enable building OTLP GRPC exporter.
    - `-DWITH_OTLP_HTTP=ON` : To enable building OTLP HTTP exporter.
    - `-DWITH_PROMETHEUS=ON` : To enable building prometheus exporter.
+   - `-DOPENTELEMETRY_INSTALL=ON`: To install `otel-cpp` library needed for external code linking.
```
Comment (marcalff) [inline INSTALL.md:88]:
```suggestion
   - `-DOPENTELEMETRY_INSTALL=ON`: To install `otel-cpp` library needed
      for external code linking.
```

**Review by marcalff — APPROVED** [review]
Thanks for the documentation fix.

**Pull request merged**